core/tx_pool.go

```diff
@@ -1280,6 +1289,32 @@ func (pool *TxPool) reset(oldHead, newHead *types.Header) {
 	pool.istanbul = pool.chainconfig.IsIstanbul(next)
 }
 
+// This re-validates accounts which have been touched in a recent block
+func (pool *TxPool) validateAAExecutables(txs []*types.Transaction) {
+	for _, tx := range txs {
+		var account = tx.To()
+		if account != nil && tx.IsAA() {
+			if pool.Has(tx.Hash()) {
+				pool.removeTx(tx.Hash(), true)
+			} else {
+				if pending := pool.pending[*account]; pending != nil {
+					pending_tx := pending.txs.Flatten()[0]
+					pool.removeTx(pending_tx.Hash(), true)
+					// ignore locals for now
+					pool.add(pending_tx, false)
+				}
+
+				if queued := pool.queue[*account]; queued != nil {
```

Comment:
Wouldn't an `else if` be appropriate here? If I recall correctly, we currently only allow one pending or one queued transaction. Also, if there was a pending tx before, removing it and re-adding it would put it into the queue, no? So it would then be removed and re-added again here.

Reply:
yeah, you're right. When we expand to multiple txs we'll need to think about this further as well.